godot-core/src/init/mod.rs

```diff
@@ -53,17 +55,53 @@ pub unsafe fn __gdext_load_library<E: ExtensionLibrary>(
     is_success.unwrap_or(0)
 }
 
+static LOADED_LEVEL: AtomicU8 = AtomicU8::new(0);
+
+const LOADED_LEVEL_NONE: u8 = 0;
+const LOADED_LEVEL_CORE: u8 = 1;
+const LOADED_LEVEL_SERVERS: u8 = 2;
+
 unsafe extern "C" fn ffi_initialize_layer<E: ExtensionLibrary>(
     _userdata: *mut std::ffi::c_void,
     init_level: sys::GDExtensionInitializationLevel,
 ) {
     let level = InitLevel::from_sys(init_level);
     let ctx = || format!("failed to initialize GDExtension level `{:?}`", level);
 
-    // Swallow panics. TODO consider crashing if gdext init fails.
-    let _ = crate::private::handle_panic(ctx, || {
+    fn try_load<E: ExtensionLibrary>(level: InitLevel) {
+        // Workaround for https://github.com/godot-rust/gdext/issues/629:
+        // When using editor plugins, Godot may unload all levels but only reload from Scene upward.
+        // Manually run initialization of lower levels.
+        match (level, LOADED_LEVEL.load(Relaxed)) {
+            (InitLevel::Servers, LOADED_LEVEL_NONE) => {
+                try_load::<E>(InitLevel::Core);
+            }
+            (InitLevel::Scene, LOADED_LEVEL_NONE | LOADED_LEVEL_CORE) => {
+                try_load::<E>(InitLevel::Servers);
+            }
```

Comment:
What do you think about adding warning or verbose logging on here? Since In common scenario(or, during ideal hot reloading) this routine should not be executed, I think it'd be better to notify user that the workaround path was executed. It'll give some hint when any kind of side-effect occurs due to this workaround.

Reply:
And .. if so, may I use `godot_warn!` macro here?

Reply:
But in case of editor plugins, this would always be executed, no?

A warning would indicate that the user could do something differently, but I'm not sure that is the case?

Reply:
It seems these are only executed after hot reloading. Since this PR is just a workaround for upstream behavior which doesn't call necessary initialization steps, it seems better to regard this logic is 'broken' when any issue occurs after hot reloading.
Maybe it'd help users to identify any experiencing issue is related with hot-reloading, and then motivate them to include the log meesage to issue report...?

I'm not entirely sure for this. Also I agree that there's nothing that user can deal with this.